rust/core-lib/src/editor.rs

```diff
@@ -419,8 +419,8 @@ impl<W: Write + Send + 'static> Editor<W> {
     }
 
     pub fn merge_new_state(&mut self, new_engine: Engine) {
-        // TODO: CRDT merge
-        self.engine = new_engine;
+        print_err!("Xi: merging new state from ledger");
```

Comment:
I think this is fine for now but it's leaving me wishing for a more systematic way to introspect into the core's behavior.